pkg/shell/cockpit-storage.js

```diff
@@ -1845,44 +1845,39 @@ PageStorageDetail.prototype = {
 
         is_running = !!block;
 
+        state = is_running? _("Running") : _("Not running");
+
         if (raid.Degraded > 0) {
-            degraded = ('<span style="color:red">' + _("ARRAY IS DEGRADED") + '</span> -- ' +
+            state = ('<span style="color:red">' + _("ARRAY IS DEGRADED") + '</span> -- ' +
                         cockpit.format(_("%0 disks are missing"), raid.Degraded));
-        }
-        if (!raid.SyncAction) {
-            if (block) {
-                state = _("Running");
+        } else if (is_running) {
+            state = _("Running");
+        } else {
+            state = _("Not running");
+        }
+
+        action = { "idle" : "",
+                   "check" : _("Data Scrubbing"),
+                   "repair" : _("Data Scrubbing and Repair"),
+                   "resync" : _("Resyncing"),
+                   "recover" : _("Recovering "),
+                   "frozen" : _("Frozen")
+                 }[raid.SyncAction] || raid.SyncAction;
```

Comment:
I would suggest building this map once, and not every time this function is called.